lib/tasks/curation_tt.rake

```diff
@@ -3,117 +3,44 @@ namespace :curate do
-          (tt.name.include? "#{tgt_species} subsp. #{tgt_subtaxa}")
-        }
-        if candidate_tts.length == 1
-          pl.taxonomy_term = candidate_tts.first
-          pl.save
-          @success_cntr += 1
-          @cntr += 1
-          next pl
-        else
-          fail("FAIL #{@cntr.to_s}: Ambiguous record: #{pl_full_name}. Subtaxa #{tgt_subtaxa} duplicates existing species name. Giving up.")
-          next pl
-        end
-      end
-      if meaningful(tgt_subtaxa)
-        # Look for subtaxa in TT
-        candidate_tts = TaxonomyTerm.all.select{|tt| tt.name.include? tgt_subtaxa}
-        if candidate_tts.length == 1
-          pl.taxonomy_term = candidate_tts.first
-          pl.save
-          @success_cntr += 1
-          @cntr += 1
-          next pl
-        elsif candidate_tts.length == 0
-          fail("FAIL #{@cntr.to_s}: No candidate TT found for #{pl_full_name}.")
-          next pl
-        else
-          fail("FAIL #{@cntr.to_s}: More than one candidate TT found for for #{pl_full_name}.")
-          next pl
-        end
-      end
-
-      # tgt_subtaxa not meaningful or absent - try to assign TT by species
-      if meaningful(tgt_species) and all_genera.include? tgt_species
-        fail("FAIL #{@cntr.to_s}: Ambiguous record: #{pl_full_name}. Species #{tgt_species} duplicates existing genus name. Giving up.")
-        next pl
-      end
-      if meaningful(tgt_species)
-      # Try to find a corresponding record in TTs. Assume genus is always Brassica.
-        candidate_tts = TaxonomyTerm.where("name = 'Brassica #{tgt_species}'").all
-        if candidate_tts.length == 1
-          pl.taxonomy_term = candidate_tts.first
-          pl.save
-          @success_cntr += 1
-          @cntr += 1
-          next pl
-        elsif candidate_tts.length == 0
-          fail("FAIL #{@cntr.to_s}: No candidate TT found for #{pl_full_name}.")
-          next pl
-        else
-          fail("FAIL #{@cntr.to_s}: More than one candidate TT found for #{pl_full_name}.")
-          next pl
-        end
-      end
-
-      # tgt_species also not meaningful or absent - try to assign to Brassica sp.
-      if meaningful(tgt_genus)
-        pl.taxonomy_term = TaxonomyTerm.find_by(name: 'Brassica sp.')
+      name_to_find = if !meaningful(tgt_species)
+                       "#{tgt_genus}"
+                     elsif !meaningful(tgt_subtaxa)
+                       "#{tgt_genus} #{tgt_species}"
+                     else
+                       ["#{tgt_genus} #{tgt_species} #{tgt_subtaxa}",
+                        "#{tgt_genus} #{tgt_species} var. #{tgt_subtaxa}",
+                        "#{tgt_genus} #{tgt_species} subsp. #{tgt_subtaxa}"]
+                     end
+
+      cntr += 1
+      tt = TaxonomyTerm.find_by(name: name_to_find)
+      if tt
+        pl.taxonomy_term = tt
         pl.save
-        @success_cntr += 1
-        @cntr += 1
-        next pl
+        success_cntr += 1
```

Comment:
This task takes a long time to execute. Consider adding some on-the-fly progress messages to indicate how things are going (at least show how many records have been processed, e.g. `if @cntr % 1000 == 0 puts "Processed #{cntr.to_s} records..." end`).